cli-plugins/manager/metadata.go

```diff
@@ -23,3 +29,29 @@ type Metadata struct {
 	// URL is a pointer to the plugin's homepage.
 	URL string `json:",omitempty"`
 }
+
+// validateMetadata returns an error if any fields are missing or invalid for the
+// specified SchemaVersion.
+func validateMetadata(meta *Metadata) error {
+	if meta.SchemaVersion == "" {
+		return fmt.Errorf("%q field is required", "SchemaVersion")
+	}
+
+	switch meta.SchemaVersion {
+	case "0.1.0":
+		// clear fields not supported by version
+		meta.Name = ""
+	case "0.2.0":
+		if meta.Name == "" {
```

Comment:
We should be careful with this one (yes, I've been looking at that gnarly "must be 0.1.0" as well), but if we'd ship a plugin with 0.2.0, it would mean it cannot be used with any CLI other than docker 25.0.0

Which also means that anyone installing an older version of the deb/rpm packages will now be broken.

Reply:
Gah! You're absolutely right.

Reply:
I guess the `0.1.0` threw me for a loop and made me think "oh, semver, I should update this!"

I suppose from the perspective of a JSON-ish API, field additions are typically acceptable from a compatibility standpoint. How do you feel if I rollback the `SchemaVersion` part? Keep it as `0.1.0` but start sending `Name`.

Reply:
A possible alternative: change the signature of `plugin.Run()` to be `(name, meta, factoryFn)`. It avoids changing `Metadata` at all at the expense changing the API. (That will guarantee that all plugins start reporting UA once they upgrade at least...)

Reply:
So, I think we can change the code to remove the fixed requirement (we can set a minimum version of v0.1.0)

That won't fix existing cli's, but at least removes the weird bit for future.

Other than that, I think it's fine to add new fields, which means that we can detect if new fields are set (older CLI's would ignore them), and we could use that for feature detection.

(possibly a field that indicates what features are supported, if that helps avoid ambiguity between "empty field" and "feature supported but some field is empty")